packages/core/src/enqueue_links/shared.ts

```diff
@@ -1,68 +1,149 @@
-            pUrl = item;
-        } else if (typeof item === 'string' || item instanceof RegExp) { // If it's a string or RegExp, construct a PURL from it directly.
-            pUrl = new PseudoUrl(item);
-        } else { // If it's an object, look for a purl property and use it and the rest to construct a PURL with a Request template.
-            const { purl, ...opts } = item;
-            pUrl = new PseudoUrl(purl, opts);
+        // Get pseudoUrl object from cache.
+        let regexpObject = enqueueLinksPatternCache.get(item);
+        if (regexpObject) return regexpObject;
+
+        if (typeof item === 'string') {
+            regexpObject = { regexp: purlToRegExp(item) };
+        } else {
+            const { purl, ...requestOptions } = item;
+            regexpObject = { regexp: purlToRegExp(purl), ...requestOptions };
+        }
+
+        updateEnqueueLinksPatternCache(item, regexpObject);
+
+        return regexpObject;
+    });
+}
+
+/**
+ * Helper factory used in the `enqueueLinks()` and enqueueLinksByClickingElements() function
+ * to construct Glob objects from Glob pattern strings.
+ * @ignore
+ */
+export function constructGlobObjectsFromGlobs(globs: GlobInput[]): GlobObject[] {
+    return globs.map((item) => {
+        // Get glob object from cache.
+        let globObject = enqueueLinksPatternCache.get(item);
+        if (globObject) return globObject;
+
+        if (typeof item === 'string') {
+            globObject = { glob: validateGlobPattern(item) };
+        } else {
+            const { glob, ...requestOptions } = item;
+            globObject = { glob: validateGlobPattern(glob), ...requestOptions };
         }
 
-        // Manage cache
-        enqueueLinksPseudoUrlCache.set(item, pUrl);
-        if (enqueueLinksPseudoUrlCache.size > MAX_ENQUEUE_LINKS_CACHE_SIZE) {
-            const key = enqueueLinksPseudoUrlCache.keys().next().value;
-            enqueueLinksPseudoUrlCache.delete(key);
+        updateEnqueueLinksPatternCache(item, globObject);
+
+        return globObject;
+    });
+}
+
+export function validateGlobPattern(glob: string): string {
+    const globTrimmed = glob.trim();
+    if (globTrimmed.length === 0) throw new Error(`Cannot parse Glob pattern '${globTrimmed}': it must be an non-empty string`);
+    return globTrimmed;
+}
+
+/**
+ * Helper factory used in the `enqueueLinks()` and enqueueLinksByClickingElements() function
+ * to check RegExps input and return valid RegExps.
+ * @ignore
+ */
+export function constructRegExpObjectsFromRegExps(regexps: RegExpInput[]): RegExpObject[] {
+    return regexps.map((item) => {
+        // Get regexp object from cache.
+        let regexpObject = enqueueLinksPatternCache.get(item);
+        if (regexpObject) return regexpObject;
+
+        if (item instanceof RegExp) {
+            regexpObject = { regexp: item };
+        } else {
+            const { regexp, ...requestOptions } = item;
+            regexpObject = { regexp, ...requestOptions };
```

Comment:
Isn't this a bit redundant? Or did you mean to create a RegExp from the deconstructed one?

Reply:
It is redundant - went from globs from above and just used the same pattern. So you're right, it's not necessary here.